src/fsmc_lcd/mod.rs

```diff
@@ -0,0 +1,410 @@
+pub struct FsmcLcd<PINS> {
+    pins: PINS,
+    fsmc: FSMC,
+}
+
+impl<PINS> FsmcLcd<PINS>
+where
+    PINS: Pins,
+{
+    /// Configures the FSMC/FMC to interface with an LCD using the provided pins
+    ///
+    /// The same timings will be used for all connected LCDs.
+    ///
+    /// The return type includes an `FsmcLcd` and either an `Lcd` or a tuple of up to four `Lcd`s,
+    /// depending on the pins given.
+    ///
+    /// The returned `FsmcLcd` can be used later to release the FSMC and pins for other uses,
+    /// or it can be ignored.
+    ///
+    /// # Return type examples
+    ///
+    /// ## One enable/chip select pin
+    ///
+    /// If you pass an `LcdPins` object with the `enable` field containing a single `ChipSelect1`
+    /// object with a pin, this function will return an `FsmcLcd` and an `Lcd<SubBank1>`.
+    ///
+    /// ### Multiple enable/chip select pins
+    ///
+    /// If you pass an `LcdPins` object with the `enable` field containing a tuple of 2-4
+    /// `ChipSelectX` objects, this function will return an `FsmcLcd` and a tuple of `Lcd<_>`
+    /// objects. Each `Lcd` is associated with one chip select pin, and can be controlled
+    /// independently.
+    ///
+    pub fn new(
+        fsmc: FSMC,
+        pins: PINS,
+        read_timing: &Timing,
+        write_timing: &Timing,
+    ) -> (Self, PINS::Lcds) {
+        use self::sealed::Conjure;
+        unsafe {
+            //NOTE(unsafe) this reference will only be used for atomic writes with no side effects
+            let rcc = &(*RCC::ptr());
+            // Enable the FSMC/FMC peripheral
+            // All STM32F4 models with an FSMC or FMC use bit 0 in AHB3ENR and AHB3RSTR.
+            bb::set(&rcc.ahb3enr, 0);
+
+            // Stall the pipeline to work around erratum 2.1.13 (DM00037591)
+            cortex_m::asm::dsb();
+            // Reset FSMC/FMC
+            bb::set(&rcc.ahb3rstr, 0);
+            bb::clear(&rcc.ahb3rstr, 0);
+        }
+
+        // Configure memory type and basic interface settings
+        // The reference manuals are sometimes unclear on the distinction between banks
+        // and sub-banks of bank 1. This driver uses addresses in the different sub-banks of
+        // bank 1. The configuration registers for "bank x" (like FMC_BCRx) actually refer to
+        // sub-banks, not banks. We need to configure and enable all four of them.
+        configure_bcr1(&fsmc.bcr1);
+        configure_bcr(&fsmc.bcr2);
+        configure_bcr(&fsmc.bcr3);
+        configure_bcr(&fsmc.bcr4);
+        configure_btr(&fsmc.btr1, read_timing);
+        configure_btr(&fsmc.btr2, read_timing);
+        configure_btr(&fsmc.btr3, read_timing);
+        configure_btr(&fsmc.btr4, read_timing);
+        configure_bwtr(&fsmc.bwtr1, write_timing);
+        configure_bwtr(&fsmc.bwtr2, write_timing);
+        configure_bwtr(&fsmc.bwtr3, write_timing);
+        configure_bwtr(&fsmc.bwtr4, write_timing);
```

Comment:
That reasoning seems to be not quite correct. As to my interpretation, the configuration of the sub-banks of bank 1 (NOR/PSRAM controller) overlaps with the other banks 2-4 (NAND/PC Card subbanks). That does not mean you need to configure all of them (and indeed we may want to allow other users of FSMC in the future in which case it would cause major conflicts).

Since the common connection between the banks and the sub-banks is the NOE/CS pin we probably can use them as universal owned resource indicating the use of a bank/PSRAM sub-bank and thus tie the configuration of banks to them.